programs/nftoken/src/ix_mintlist_add_mint_infos.rs

```diff
@@ -0,0 +1,49 @@
+use crate::account_types::*;
+use crate::errors::*;
+use anchor_lang::prelude::*;
+use std::convert::TryInto;
+
+/// Adds multiple `MintInfo`'s to the `mintlist`.
+pub fn mintlist_add_mint_infos_inner(
+    ctx: Context<MintlistAddMintInfos>,
+    mint_infos: Vec<MintInfoArg>,
+) -> Result<()> {
+    let mintlist = &mut ctx.accounts.mintlist;
+
+    let len_to_add: u16 = mint_infos
+        .len()
+        .try_into()
+        .map_err(|_err| error!(NftokenError::TooManyMintInfos))?;
+
+    let mint_infos_added = mintlist.mint_infos_added;
+    let available_mint_info_slots = mintlist.num_mints - mint_infos_added;
+
+    require!(
+        len_to_add as u16 <= available_mint_info_slots,
+        NftokenError::TooManyMintInfos
+    );
+
+    mintlist.mint_infos_added += len_to_add;
+
+    let mintlist_account_info = mintlist.to_account_info();
+    let mut mintlist_data = mintlist_account_info.data.borrow_mut();
+
+    let mint_info_size = MintInfo::size();
+    let insertion_byte_pos = MintlistAccount::size(mint_infos_added);
+    for (i, mint_info_arg) in mint_infos.iter().enumerate() {
+        let mint_info = MintInfo::from(mint_info_arg);
+        mint_info.serialize(&mut &mut mintlist_data[insertion_byte_pos + i * mint_info_size..])?
+        // mintlist_account.mint_infos[usize::from(mint_infos_added) + i] = mint_info.into()
```

Comment:
@vovacodes are we still using this? I'm guessing it didn't work so we can remove it?

```suggestion
```

Reply:
Ah right, just forgot to remove it